base/iterators.jl

```diff
@@ -1460,6 +1463,35 @@ julia> unfold(6) do n
  1
  1
 ```
+
+# Extended help
+
+The interface for `f` is very similar to the interface required by `iterate`, but `unfold` is simpler to use because it does not require you to define a type.
```

Comment:
Isn't the interface exactly the same?

Reply:
`iterate` receives a struct as the first argument, whereas `f` just receives the state, so it's a bit different. If you define `iterate` you have to provide two methods instead of one as well:

```julia
iterate(itr::YourType) = # ...
iterate(itr::YourType, state) = # ...
```

The return type is the same, though.